src/test/java/core/basesyntax/BlockingQueueTest.java

```diff
@@ -0,0 +1,74 @@
+package core.basesyntax;
+
+import core.basesyntax.thread.Consumer;
+import core.basesyntax.thread.Producer;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class BlockingQueueTest {
+    private BlockingQueue<Integer> blockingQueue;
+
+    @Test
+    public void put_emptySpaceLeft_ok() throws InterruptedException {
+        blockingQueue = new BlockingQueue<>(100);
+        Thread producer = new Thread(new Producer(blockingQueue));
+
+        producer.start();
+        Thread.sleep(500);
+        Assert.assertNotEquals("Your `put()` method should not put the producer thread "
+                + "in the WAITING state if there is still space in the queue",
+                Thread.State.WAITING, producer.getState());
+    }
+
+    @Test
+    public void put_fullCapacity_ok() throws InterruptedException {
+        blockingQueue = new BlockingQueue<>(49);
+        Thread producer = new Thread(new Producer(blockingQueue));
+        Thread consumer = new Thread(new Consumer(blockingQueue));
+
+        producer.start();
+        Thread.sleep(1000);
+        Assert.assertEquals("Your `put()` method should put the producer thread "
+                + "in the WAITING state if there is no space in the queue",
+                Thread.State.WAITING, producer.getState());
+        consumer.start();
+        Thread.sleep(100);
+        Assert.assertNotEquals("Your producer thread should continue working "
+                + "if some space appeared in the queue",
+                Thread.State.WAITING, producer.getState());
+    }
+
+    @Test
+    public void take_ok() throws InterruptedException {
+        blockingQueue = new BlockingQueue<>(50);
```

Comment:
Is the capacity parameter required? We haven't test cases to verify the behavior that happens if capacity is reached.
```suggestion
        blockingQueue = new BlockingQueue<>();
```

Reply:
discussed, thanks